modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/utils/BlobsExtractor.java

```diff
@@ -182,8 +185,14 @@ protected void findBlobsProperties(Property property, List<String> split, List<P
                 properties.add(property);
             }
         } else {
-            for (Property childProperty : property.getChildren()) {
-                Property childSubProp = childProperty.get(split.get(0));
+            if (property.isList()) {
+                for (Property childProperty : property.getChildren()) {
+                    Property childSubProp = childProperty.get(split.get(0));
+                    List<String> subPath = split.subList(1, split.size());
+                    findBlobsProperties(childSubProp, subPath, properties);
+                }
+            } else { // complex type
+                Property childSubProp = property.get(split.get(0));
                 List<String> subPath = split.subList(1, split.size());
```

Comment:
`split.get(0)` and `List<String> subPath` could be factored out of the `if` and loop.